README.md

```diff
@@ -1 +1,73 @@
-# miden-node
\ No newline at end of file
+# Miden node
+
+<a href="https://github.com/0xPolygonMiden/miden-node/blob/main/LICENSE"><img src="https://img.shields.io/badge/license-MIT-blue.svg"></a>
+<img src="https://github.com/0xPolygonMiden/miden-node/workflows/CI/badge.svg?branch=main">
+<a href="https://crates.io/crates/miden-node"><img src="https://img.shields.io/crates/v/miden-node"></a>
+
+This repository holds the Miden node; that is, the software which processes transactions and creates blocks for the Miden rollup.
+
+### Status
+
+The Miden node is still under heavy development and the project can be considered to be in an *alpha* stage. Many features are yet to be implemented and there is a number of limitations which we will lift in the near future.
+
+At this point, we are developing the Miden node for a centralized operator. Thus, the work does not yet include such components as P2P networking and consensus. These also will be added in the future.
+
+## Architecture
+
+The Miden node is made up of 3 main components: 
+- **store:** stores the current state of the chain.
+- **rpc:** serves client requests such as to synchronize with the latest state of the chain or to submit transactions.
+- **block producer:** accepts transactions from the RPC component, creates blocks containing those transactions, and sends them to the store.
+
+## Usage
+
+Before running the node, you must first generate the genesis file. 
+
+### Generating the genesis file
+
+The contents of the genesis file are currently hardcoded in Rust, but we intend to make those configurable shortly. The genesis block currently sets up 2 accounts: a faucet account for a `POL` token, as well as a wallet account.
+
+To generate the file for production, run 
+
+```sh
+$ cargo run -p miden-node -- make-genesis
+```
```

Comment:
I wonder if a better approach would be to split compiling and other commands into separate steps. Basically:

- Step 1: compile Miden node using `cargo build`. Then we get `miden-node` executable.
- Other steps: run `miden-node` executable with various commands.

Reply:
Agreed - done